lib/temporal/activity/task_processor.rb

```diff
@@ -26,16 +29,17 @@ def process
           raise ActivityNotRegistered, 'Activity is not registered with this worker'
         end
 
-        metadata = Metadata.generate(Metadata::ACTIVITY_TYPE, task, namespace)
         context = Activity::Context.new(client, metadata)
 
-        result = middleware_chain.invoke(metadata) do
+        result = middleware_chain.invoke(metadata, task.to_h) do
```

Comment:
If I had been able to see the task alongside the exception we saw then it would have been immediately obvious what the issue was (query being sent with no workflow tasks). Maybe I'm bias as an SDK contributor, but this information is extremely valuable to me.